src/Me/Modals/MentorshipRequest.js

```diff
@@ -0,0 +1,175 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { Modal } from './Modal';
+import FormField from '../components/FormField';
+import Textarea from '../components/Textarea';
+import { toast } from 'react-toastify';
+import { sendMentorshipRequest, getMentorshipRequest } from '../../api';
+import messages from '../../messages';
+import ImageSrc from '../../assets/mentorshipRequestSuccess.svg';
+
+const MentorshipRequestDetails = styled.div`
+  margin: 0 auto;
+  width: 100%;
+  max-width: 52rem;
+`;
+
+const MentorshipRequestDetailsForm = styled.form`
+  display: flex;
+  width: 100%;
+`;
+
+const SuccessMentorshipRequest = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+
+  p {
+    max-width: 24rem;
+  }
+`;
+
+const FormFields = styled.div`
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: space-between;
+  align-items: stretch;
+  padding: 0 5px;
+  width: 100%;
+`;
+
+const ExtendedFormField = styled(FormField)`
+  flex: 1 1 100%;
+  width: 100%;
+  min-width: 15rem;
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-start;
+
+  & label {
+    color: #4f4f4f;
+  }
+`;
+
+const MentorshipRequest = ({ mentor, closeModal }) => {
+  const [confirmed, setConfirmed] = useState(false);
+  const [mentorshipRequestDetails, setMentorshipRequestDetails] = useState(
+    getMentorshipRequest()
+  );
+
+  // text fields onChange function
+  const handleInputChange = e => {
+    const {
+      target: { name: fieldName, value },
+    } = e;
+    setMentorshipRequestDetails({
+      ...mentorshipRequestDetails,
+      [fieldName]: value,
+    });
+  };
+
+  const model = {
+    message: {
+      label: 'Message',
+      type: 'longtext',
+      defaultValue: mentorshipRequestDetails.message,
+      validate: value => !!value,
+      required: true,
+      placeholder: 'Anything else you want to say?',
+    },
```

Comment:
Sorry got going back and forth. Crysfel asked to add also background and my expectations. Is that ok?

Reply:
ok i'll add those in with the payload as well.